experiment/measurer/measure_manager.py

```diff
@@ -755,51 +700,294 @@ def get_pool_args(measurers_cpus, runners_cpus):
+            if unmeasured_snapshot_identifier not in queued_snapshots:
+                self.put_task_in_request_queue(unmeasured_snapshot,
+                                               request_queue)
+                queued_snapshots.add(unmeasured_snapshot_identifier)
+
+        # Read results from response queue.
+        measured_snapshots = self.consume_snapshots_from_response_queue(
+            response_queue, queued_snapshots)
+        logger.info('Retrieved %d measured snapshots from response queue.',
+                    len(measured_snapshots))
+
+        # Save measured snapshots to database.
+        if measured_snapshots:
+            db_utils.add_all(measured_snapshots)
 
+        return True
+
+    def measure_manager_loop(self, max_total_time: int):
+        """Measure manager loop. Creates request and response queues, request
+        measurements tasks from workers, retrieve measurement results from
+        response queue and writes measured snapshots in database."""
+        logger.info('Starting measure manager loop.')
+        with multiprocessing.Pool() as pool:
+            set_up_coverage_binaries(pool, self.experiment)
+        (request_queue, response_queue) = self.initialize_queues()
+        self.start_workers(request_queue, response_queue)
         max_cycle = _time_to_cycle(max_total_time)
         queued_snapshots = set()
-        while not scheduler.all_trials_ended(experiment):
-            continue_inner_loop = measure_manager_inner_loop(
-                experiment, max_cycle, request_queue, response_queue,
-                queued_snapshots)
+        while not scheduler.all_trials_ended(self.experiment):
+            continue_inner_loop = self.measure_manager_inner_loop(
+                max_cycle, request_queue, response_queue, queued_snapshots)
             if not continue_inner_loop:
                 break
             time.sleep(MEASUREMENT_LOOP_WAIT)
         logger.info('All trials ended. Ending measure manager loop')
 
 
+class LocalMeasureManager(BaseMeasureManager):
+    """Class that holds implementations of core methods for running a measure
+    worker locally."""
+
+    def __init__(self,
+                 experiment: str,
+                 region_coverage=False,
+                 measurers_cpus=None):
+        super().__init__(experiment, region_coverage)
+        self.measurers_cpus = measurers_cpus
+
+    def initialize_queues(
+        self
+    ) -> Tuple[multiprocessing.queues.Queue, multiprocessing.queues.Queue]:
+        return (multiprocessing.Queue(), multiprocessing.Queue())
+
+    def start_workers(self, request_queue: multiprocessing.queues.Queue,
+                      response_queue: multiprocessing.queues.Queue):
+        if not self.measurers_cpus:
+            self.measurers_cpus = multiprocessing.cpu_count()
+            logger.info(
+                'Number of measurer CPUs not passed as argument. using %d',
+                self.measurers_cpus)
```

Comment:
Would it be simpler to set `self. measurers_cpus` in the `__init__()` of `BaseMeasureManager`, given `GoogleCloudMeasureManager` does the same?

Reply:
Yes! it would, I did it this way because I don't plan to keep this logic on the google cloud measure manager for too long.

Later, the google cloud measure manager won't spawn processes, but rather, it will start a pool of autoscaling VMs to consume from the queue. So I figured it would be better to keep the two logics separated, even though they are the same now